### PR TITLE
Calculate version range in init-compiler.sh script

### PR DESCRIPTION
**Pull request by am11**

### CODE REVIEW

File: eng/common/native/init-compiler.sh [hunk 1]
```diff
@@ -84,43 +82,28 @@ if [ -z "$CLR_CC" ]; then
...
-        # note: gcc (all versions) and clang versions higher than 6 do not have minor version in file name, if it is zero.
-        if [ "$compiler" = "clang" ]; then versions="19 18 17 16 15 14 13 12 11 10 9 8 7 6.0 5.0 4.0 3.9 3.8 3.7 3.6 3.5"
-        elif [ "$compiler" = "gcc" ]; then versions="14 13 12 11 10 9 8 7 6 5 4.9"; fi
-
-        for version in $versions; do
-            _major="${version%%.*}"
-            [ -z "${version##*.*}" ] && _minor="${version#*.}"
-            desired_version="$(check_version_exists "$_major" "$_minor")"
-            if [ "$desired_version" != "-1" ]; then majorVersion="$_major"; break; fi
+        i="$((maxVersion + 1))" # +1 for headspace
```
Comment (am11) [inline eng/common/native/init-compiler.sh:85]:
```suggestion
        maxVersion="$((maxVersion + 1))" # +1 for headspace
        i="$maxVersion"
```

File: eng/common/native/init-compiler.sh [hunk 2]
```diff
@@ -84,43 +82,29 @@ if [ -z "$CLR_CC" ]; then
...
-        elif [ "$compiler" = "gcc" ]; then versions="14 13 12 11 10 9 8 7 6 5 4.9"; fi
-
-        for version in $versions; do
-            _major="${version%%.*}"
-            [ -z "${version##*.*}" ] && _minor="${version#*.}"
-            desired_version="$(check_version_exists "$_major" "$_minor")"
-            if [ "$desired_version" != "-1" ]; then majorVersion="$_major"; break; fi
+        maxVersion="$((maxVersion + 1))" # +1 for headspace
+        i="$maxVersion"
+        minVersion=8
```
Comment (am11) [inline eng/common/native/init-compiler.sh:87]:
```suggestion
        minVersion=8
        maxVersion="$((maxVersion + 1))" # +1 for headspace
        i="$maxVersion"
```

**Review by janvorli — APPROVED** [review]
LGTM, thank you

**Review by akoeplinger — APPROVED** [review]

**Comment by jkotas** [comment]
This change introduced sudden build break: https://github.com/dotnet/runtime/issues/105784

Also, this change goes against principles for deterministic builds: We want the build in a given image to keep working the same way irrespective of the current time.

**Comment by am11** [comment]
> Also, this change goes against principles for deterministic builds: We want the build in a given image to keep working the same way irrespective of the current time.

It was discussed in https://github.com/dotnet/runtime/pull/105129 and it doesn't work that way. We validated it against the CI including wasm leg. That's something which didn't happen during the PR run. I can look into it later today.

**Pull request merged**